### PR TITLE
Send full registration in case server response NotFound for the regis…

### PR DESCRIPTION
**Pull request by anttiylitokola**
…tration update message

### CODE REVIEW

File: source/m2mnsdlinterface.cpp [hunk 1]
```diff
@@ -451,12 +451,18 @@ uint8_t M2MNsdlInterface::received_from_server_callback(struct nsdl_s * /*nsdl_h
...
                 M2MInterface::Error error = interface_error(coap_header);
+                // Send full registration in case server has lost the endpoint information
+                if (coap_header->msg_code == COAP_MSG_CODE_RESPONSE_NOT_FOUND) {
+                    _register_id = sn_nsdl_register_endpoint(_nsdl_handle,_endpoint);
+                }
                 // In case, the error for update registration is not allowed implies the client is no longer registered in
                 // server hence the error returns should be NotRegistered.
-                if(M2MInterface::NotAllowed == error) {
-                    error = M2MInterface::NotRegistered;
+                else {
```
Comment (yogpan01) [inline source/m2mnsdlinterface.cpp]: This check should be removed, client should do full registration if there is any error coming on Registration Update

**Comment by anttiylitokola** [comment]
rebuild

**Comment by anttiylitokola** [comment]
rebuild

**Comment by anttiylitokola** [comment]
rebuild

**Comment by yogpan01** [comment]
LGTM

**Pull request merged**